code/modules/mob/living/simple_animal/hostile/terror_spiders/objects.dm

```diff
@@ -0,0 +1,455 @@
+				var/mob/living/simple_animal/hostile/poison/terror_spider/S = new grow_as(loc)
+				S.faction = faction
+				S.spider_myqueen = spider_myqueen
+				S.master_commander = master_commander
+				S.ai_playercontrol_allowingeneral = ai_playercontrol_allowingeneral
+				S.enemies = enemies
+				if (spider_queen_declared_war)
+					S.spider_queen_declared_war = 1
+					if (!istype(S, /mob/living/simple_animal/hostile/poison/terror_spider/purple))
+						S.idle_ventcrawl_chance = 15
+				qdel(src)
+
+// --------------------------------------------------------------------------------
+// ----------------- TERROR SPIDERS: PROJECTILES ----------------------------------
+// --------------------------------------------------------------------------------
+
+
+/obj/item/projectile/terrorqueenspit
+	name = "poisonous spit"
+	damage = 0
+	icon_state = "toxin"
+	damage_type = TOX
+
+
+/obj/item/projectile/terrorqueenspit/on_hit(var/mob/living/carbon/target)
+	if (istype(target, /mob))
+		var/mob/living/L = target
+		if (L.reagents)
+			if (L.can_inject(null,0,"chest",0))
+				L.reagents.add_reagent("terror_queen_toxin",15)
+		if (!istype(L, /mob/living/simple_animal/hostile/poison/terror_spider))
+			L.adjustToxLoss(40) // Terror Spiders are immune to their queen's spit.
+
+
+/obj/item/projectile/terrorempressspit
+	name = "poisonous spit"
+	damage = 0
+	icon_state = "toxin"
+	damage_type = TOX
+
+
+/obj/item/projectile/terrorempressspit/on_hit(var/mob/living/carbon/target)
+	if (istype(target, /mob))
+		var/mob/living/L = target
+		if (L.reagents)
+			L.reagents.add_reagent("ketamine",30)
+		if (!istype(L, /mob/living/simple_animal/hostile/poison/terror_spider))
+			L.adjustToxLoss(60) // Terror Spiders are immune to their queen's spit.
+
+/obj/effect/spider/terrorweb
+	name = "terror web"
+	desc = "it's stringy and sticky"
+	icon = 'icons/effects/effects.dmi'
+	anchored = 1 // prevents people dragging it
+	density = 0 // prevents it blocking all movement
+	health = 20 // two welders, or one laser shot (15 for the normal spider webs)
+	icon_state = "stickyweb1"
+
+
+/obj/effect/spider/terrorweb/New()
+	if (prob(50))
+		icon_state = "stickyweb2"
+
+/obj/effect/spider/terrorweb/proc/DeCloakNearby()
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/gray/G in view(6,src))
+		if (G.stat != DEAD)
+			G.GrayDeCloak()
+			G.Aggro()
+
+/obj/effect/spider/terrorweb/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if (air_group || (height==0)) return 1
+	if (istype(mover, /mob/living/simple_animal/hostile/poison/terror_spider))
+		return 1
+	if (istype(mover, /obj/item/projectile/terrorqueenspit))
+		return 1
+	if (istype(mover, /obj/item/projectile/terrorempressspit))
+		return 1
+	if (istype(mover, /mob/living))
+		if (prob(80))
+			to_chat(mover, "<span class='danger'>You get stuck in \the [src] for a moment.</span>")
+			var/mob/living/M = mover
+			M.Stun(5) // 5 seconds.
+			M.Weaken(5) // 5 seconds.
+			M.emote("scream")
```

Comment:
The audio queue can actually be quite useful. I have watched people testrun these spiders for literally hours at a time, several times, in variable size groups, and not one single person ever has complained about the scream. In fact, many of them considered unexpectedly running into webs to be alternatively hilarious and terrifying. It was one of the best parts of the run. How it appears to you, and the experience of actual playtesters, is completely different.